pkg/git/git.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package git
+
+import (
+	"fmt"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"regexp"
+
+	gitpackage "github.com/go-git/go-git/v5"
+	//"github.com/go-git/go-git/v5/plumbing/transport/http"
+)
+
+const defaultRemote = "origin"
+var (
+	// UrlRegexFormat matches the url of SSH git repository
+	httpUrlRegexFormat = regexp.MustCompile(`^((http:\/\/)|(https:\/\/))?([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,6}(\/)`)
+	sshUrlRegexFormat = regexp.MustCompile(`(ssh:\/\/[\w\d\.]+|.+@?.+\..+:)(:[\d]+){0,1}\/*(.*)`)
+	httpsPublicUrlRegexFormat = regexp.MustCompile(`(https:\/\/)?(github\.)+com(\/)`)
+	sshPublicUrlRegexFormat = regexp.MustCompile(`(git+@github.com:)(:[\d]+){0,1}/*(.*)`)
+)
+
+func ListRemote(urlPath string) error {
```

Comment:
For me, `ListRemote` is a bit misleading solely based on the name, because it is more like checking a location, right?
```suggestion
func IsValidGitLocation(urlPath string) error {
```

Reply:
I change it from `ListRemote` to `ValidateGitURLExists`.